### PR TITLE
[DOCS] Removal of docs for deprecated POT

### PR DESCRIPTION
**Pull request by sgolebiewski-intel**
Removing articles dedicated to the deprecated tools.

### CODE REVIEW

**Review by msmykx-intel — APPROVED** [review]

File: docs/articles_en/learn_openvino/openvino_samples/c_sample_hello_classification.rst [hunk 1]
```diff
@@ -152,6 +152,5 @@ See Also
 - :doc:`Using OpenVINO™ Samples <openvino_docs_OV_UG_Samples_Overview>`
 - :doc:`Model Downloader <omz_tools_downloader>`
 - :doc:`Convert a Model <openvino_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide>`
-- :doc:`C API Reference <pot_compression_api_README>`
```
Comment (ilya-lavrenov) [inline docs/articles_en/learn_openvino/openvino_samples/c_sample_hello_classification.rst:155]: maybe we need to correct reference and keep a link to C API ?

**Comment by ilya-lavrenov** [comment]
@sgolebiewski-intel please, fix https://github.com/openvinotoolkit/openvino/actions/runs/7490344884/job/20388966270?pr=22104

File: docs/articles_en/learn_openvino/openvino_samples/c_sample_hello_classification.rst [hunk 2]
```diff
@@ -152,6 +152,4 @@ See Also
...
-
-
+- `C API Reference <https://docs.openvino.ai/2023.3/pot_compression_api_README.html>`__
```
Comment (ilya-lavrenov) [inline docs/articles_en/learn_openvino/openvino_samples/c_sample_hello_classification.rst:155]: why does it point to POT?

Comment (sgolebiewski-intel) [reply docs/articles_en/learn_openvino/openvino_samples/c_sample_hello_classification.rst]: I have updated the link with the https://github.com/openvinotoolkit/openvino/pull/22104/commits/1f4874ea0b0ee61a7924eff4caa71ed6586973de commit

Comment (sgolebiewski-intel) [inline docs/articles_en/learn_openvino/openvino_samples/c_sample_hello_classification.rst:155]:
```suggestion
- :doc:`OpenVINO Runtime C API <api/c_cpp_api/group__ov__c__api>`
```

File: thirdparty/ade
Comment (ilya-lavrenov) [inline thirdparty/ade:1]: please, remove this submodule

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**